File: caesar_cipher_encryptor/caesar_cipher_encryptor.py
def caesar_cipher_encryptor(string, key):
  new_letters = []
  alphabet = list('abcdefghijklmnopqrstuvwxyz')
  new_key = key % 26
  for letter in string:
    new_letter = get_new_letter(new_key, letter, alphabet)
    new_letters.append(new_letter)
  return "".join(new_letters)

def get_new_letter(key, letter, alphabet):
  new_key = alphabet.index(letter) + key
  return alphabet[new_key % 26]

print(caesar_cipher_encryptor('xyz', 2))

File: caesar_cipher_encryptor/caesar_cipher_encryptor.ts
function caesarCipherEncryptor(str: string, key: number): string {
    const newLetters: string[] = [];
    const newKey = key % 26;

    for (const letter of str) {
        const newLetter = getNewLetter(letter, newKey);
        newLetters.push(newLetter);
    }

    return newLetters.join('');
}

function getNewLetter(letter: string, key: number): string {
    const newCharCode = letter.charCodeAt(0) + key;
    return newCharCode <= 122 ? String.fromCharCode(newCharCode) : String.fromCharCode(96 + newCharCode % 122);
}

console.log(caesarCipherEncryptor('xyz', 2));
